src/wallet/wallet.h

```diff
@@ -709,9 +711,35 @@ class WalletRescanReserver; //forward declarations for ScanForWalletTransactions
  * A CWallet is an extension of a keystore, which also maintains a set of transactions and balances,
  * and provides the ability to create new transactions.
  */
-class CWallet final : public CCryptoKeyStore, private interfaces::Chain::Notifications
+class CWallet final : public FillableSigningProvider, private interfaces::Chain::Notifications
```

Comment:
In the spirit of this pull request it could make sense to choose composition of `FillableSigningProvider` over inheritance.

Reply:
We do that later with the rest of the restructure with the introduction of the `ScriptPubKeyMan`. I'm not sure that it makes sense to make it a member now rather than later where a bunch of things will also change.